### PR TITLE
Add Dockerization of Mempool v2 for Umbrel App Store

### PR DESCRIPTION
**Pull request by bguillaumat**
Adding a docker directory which autogenerate confiiguration for docker containers (Frontend, Backend, MySQL)

Adding a CACHE_DIR var => This allow user to choose where the cache files will be generated! (This is an improvement for using mempool as container in Umbrel)

### CODE REVIEW

**Review by wiz — CHANGES_REQUESTED** [review]

File: docker/backend/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
+FROM node:12-buster-slim AS builder
+
+WORKDIR /build
+COPY . .
+RUN apt-get update
+RUN apt-get install -y build-essential python3 pkg-config
+RUN npm ci --production
+RUN npm i typescript
```
[the remark this answers is not in the document]
Comment (bguillaumat) [reply docker/backend/Dockerfile:8]: Yeah it's needed by the `run build`
Want me to update the package.json?

File: docker/backend/start.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+CACHE_DIR=$CACHE_DIR
+
+sed -i "s/<RPC_HOST>/${RPC_HOST}/g" mempool-config.json
+sed -i "s/<RPC_PORT>/${RPC_PORT}/g" mempool-config.json
+sed -i "s/<RPC_USER>/${RPC_USER}/g" mempool-config.json
+sed -i "s/<RPC_PASS>/${RPC_PASS}/g" mempool-config.json
+sed -i "s/<ELECTRS_HOST>/${ELECTRS_HOST}/g" mempool-config.json
+sed -i "s/<ELECTRS_PORT>/${ELECTRS_PORT}/g" mempool-config.json
+sed -i "s/<MYSQL_HOST>/${MYSQL_HOST}/g" mempool-config.json
+sed -i "s/<CACHE_DIR>/${CACHE_DIR}/g" mempool-config.json
```
[the remark this answers is not in the document]
Comment (bguillaumat) [reply docker/backend/start.sh:19]: Not sure why you've added 
```
sed -i "s/__MEMPOOL_BACKEND_MAINNET_HTTP_HOST__/${__MEMPOOL_BACKEND_MAINNET_HTTP_HOST_}/g" mempool-config.json
```

File: docker/backend/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (wiz) [reply docker/backend/Dockerfile:8]: Yeah that would be cool

File: docker/backend/start.sh [hunk 3]
```diff
@@ -1,16 +1,21 @@
...
+__BITCOIN_MAINNET_RPC_USER__=${RPC_USER:=mempool}
+__BITCOIN_MAINNET_RPC_PASS__=${RPC_PASS:=mempool}
+# ELECTRUM
+__ELECTRS_MAINNET_HTTP_HOST__=${ELECTRS_HOST:=127.0.0.1}
+__ELECTRS_MAINNET_HTTP_PORT__=${ELECTRS_PORT:=50002}
+# MYSQL
+__MYSQL_HOST__=${MYSQL_HOST:=127.0.0.1}
+__MYSQL_PORT__=${MYSQL_PORT:=3306}
+
+mkdir __MEMPOOL_BACKEND_CACHE_DIR__
```
Comment (wiz) [inline docker/backend/start.sh:18]:
```suggestion
mkdir "${__MEMPOOL_BACKEND_CACHE_DIR__}"
```

**Review by wiz — APPROVED** [review]
utACK

**Pull request merged**